src/workerd/api/util.c++

```diff
@@ -69,31 +69,64 @@ void parseQueryString(kj::Vector<kj::Url::QueryParam>& query, kj::ArrayPtr<const
 kj::Maybe<kj::ArrayPtr<const char>> readContentTypeParameter(kj::StringPtr contentType,
                                                              kj::StringPtr param) {
   KJ_IF_MAYBE(semiColon, contentType.findFirst(';')) {
+    // Get to the parameters
     contentType = contentType.slice(*semiColon + 1);
-    while (contentType.startsWith(" ")) {
-      contentType = contentType.slice(1);
-    }
 
     // The attribute name of a MIME type parameter is always case-insensitive. See definition of
     // the attribute production rule in https://tools.ietf.org/html/rfc2045#page-29
-    auto lowerContentType = toLower(kj::str(contentType));
     auto lowerParam = toLower(kj::str(param));
-    if (!lowerContentType.startsWith(lowerParam)) return nullptr;
-
-    contentType = contentType.slice(param.size());
-    if (!contentType.startsWith("=")) return nullptr;
-    contentType = contentType.slice(1);
 
-    if (contentType.size() == 0) return nullptr;
-    bool quoted = contentType[0] == '"';
+    kj::StringPtr leftover = contentType;
+    while(true) {
+      while (leftover.startsWith(" ")) {
+        leftover = leftover.slice(1);
+      }
 
-    if (quoted) {
-      contentType = contentType.slice(1);
-      KJ_IF_MAYBE(endQuote, contentType.findFirst('"')) {
-        return contentType.slice(0, *endQuote);
+      KJ_IF_MAYBE(equal, leftover.findFirst('=')) {
+        // Handle parameter
+        auto name = toLower(kj::str(leftover.slice(0, *equal)));
+        auto valueStart = *equal + 1;
+        kj::ArrayPtr<const char> value = nullptr;
+
+        if (leftover[valueStart] == '"') {
+          // parameter value surrounded by quotes
+          KJ_IF_MAYBE(valueEnd, leftover.slice(valueStart + 1).findFirst('"')) {
```

Comment:
Just as a side note... (not critical for now but worth taking note of) ... this doesn't quite handle quoted values correctly. Unfortunately, every mimetype parameter is defined differently with respect to quoting. Some allow them, some don't, some follow `quoted-string` rules (which may contain quoted-pair escape sequences) and some don't. The `boundary` parameter only specifies that values may be quoted so the current impl works for those, but for other parameter values, additional processing would be required.

Reply:
Yes, you're correct in that `quoted-string` is not accounted for. But I think this is sufficient for now, as you pointed out.

Reply:
Would you mind adding a quick comment to this effect here? It's something we may eventually want to revisit.

Reply:
Done!